src/psyclone/psyir/transformations/loop_fuse_trans.py

```diff
@@ -110,6 +127,160 @@ def validate(self, node1, node2, options=None):
                         f"{node1.view()}\n"
                         f"{node2.view()}"))
 
+            vars1 = VariablesAccessInfo(node1)
+            vars2 = VariablesAccessInfo(node2)
+    
+            # Check if the loops have the same loop variable
+            loop_var1 = node1.variable
+            loop_var2 = node2.variable
+            if loop_var1 != loop_var2:
+                # If they don't have the same variable, find out if the one
+                # loop accesses the other loops variable symbol.
+                # If so, then for now we disallow this merge (though we could
+                # in theory allow using the unused one unless both use each others)
+                for signature in vars1:
+                    var_name = str(signature)
+                    if var_name == loop_var2.name:
+                        raise TransformationError(
+                            f"Error in {self.name} transformation. First "
+                            f"loop contains accesses to the second loop's "
+                            f"variable.")
```

Comment:
Maybe: `f"variable: '{var_name}'.")`

And the same below

Reply:
Added (though I think it should be `{loop_varX.name}` so I've gone with that).